### ISSUE TITLE
[PLUGIN_NAME_HERE]: <YOUR TITLE HERE>Error during Google sign-in: type 'List<Object?>' is not a subtype of type 'PigeonUserDetails?' in type cast

### ISSUE THREAD

**Issue by rajatsingh0406** [firebase/flutterfire#13388]
### Is there an existing issue for this?

- [X] I have searched the existing issues.

### Which plugins are affected?

Auth

### Which platforms are affected?

Android, iOS

### Description

My app is getting crashed when I am trying to login through google, and some time I am getting issue that the emails are also not showing

Future<void> signInWithGoogle() async {
    try {
      // begin interactive signIn process
      final GoogleSignInAccount? gUser = await GoogleSignIn(
        scopes: [
          'email',
        ],
      ).signIn();

      if (gUser == null) {
        Fluttertoast.showToast(msg: "Google sign-in canceled");
        return;
      }

      // obtaining auth details from request
      final GoogleSignInAuthentication gAuth = await gUser.authentication;

      // create a new credential for user
      final credential = GoogleAuthProvider.credential(
          accessToken: gAuth.accessToken, idToken: gAuth.idToken);

      UserCredential userCredential =
          await FirebaseAuth.instance.signInWithCredential(credential);

      User? user = userCredential.user;
      if (user != null) {
        await user.reload();
        user = FirebaseAuth.instance.currentUser;
      }

      print("this is access token----${gAuth.accessToken}");

      // Get the access token
      final accessToken = gAuth.accessToken;
      final fcmToken = await PrefManager.getUserFCMToken();

    
    } catch (e) {
      debugPrint("Error during Google sign-in: ${e.toString()}");
      Fluttertoast.showToast(msg: "Error during Google sign-in");
    }
  }

### Reproducing the issue

Clicking on google sigin button then getting the issue some time app is also crashing.

### Firebase Core version

2.24.2

### Flutter Version

3.22.0

### Relevant Log Output

```shell
I/flutter ( 4595): Error during Google sign-in: type 'List<Object?>' is not a subtype of type 'PigeonUserDetails?' in type cast
E/OplusCustomizeRestrictionManager( 4595): sInstance is null, start a new sInstance
D/CompatibilityChangeReporter( 4595): Compat change id reported: 210923482; UID 10366; state: ENABLED
D/CompatibilityChangeReporter( 4595): Compat change id reported: 147798919; UID 10366; state: ENABLED
```

### Flutter dependencies

<details>
<summary>Expand <code>Flutter dependencies</code> snippet</summary>
<br>

```yaml

Replace this line with the contents of your `flutter pub deps -- --style=compact`.

```

</details>

**Comment by SelaseKay** [comment]
Kindly bump your flutterfire plugin versions and try again

**Comment by bhagyesh-p** [comment]
I faced a similar issue where some of my Firebase packages wouldn’t update past a certain version after running `flutter pub upgrade`. Upon investigation, I realized that the issue was caused by the `cached_network_image package`.

Here’s how I resolved the problem:

 1. I removed the `cached_network_image` package from my project. 
 2. Then, I ran `flutter clean` to clear the build cache. 
 3. Next, I executed `flutter pub get` to fetch dependencies. 
 4. Finally, I ran flutter `pub upgrade` to ensure all packages were up-to-date. 


P.S. Uninstall the app on the device as well; ensure a completely clean build.
After these steps, everything returned to normal, and the Firebase packages upgraded successfully.
